fdbserver/TimedKVCache.h

```diff
@@ -0,0 +1,121 @@
+/**
+ * TimedKVCache.h
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2020 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FDBSERVER_TIMED_KVCACHE_H
+#define FDBSERVER_TIMED_KVCACHE_H
+
+#pragma once
+
+#include <list>
+#include <memory>
+#include <unordered_map>
+
+#include "flow/flow.h"
+#include "flow/Error.h"
+
+/**
+ * @class TimedKVCache
+ * @brief A key/value based cache that expires its key after a given time
+ */
+template <typename K, typename V>
+class TimedKVCache {
+public:
+	using key_t = K;
+	using value_t = V;
+
+	const double CACHE_EXPIRING_TIME_LENGTH;
+
+	explicit TimedKVCache(const double& expiringTimeLength) : CACHE_EXPIRING_TIME_LENGTH(expiringTimeLength) {}
+
+	/**
+	 * Add a new key/value pair to the cache
+	 * @param key
+	 * @param value
+	 */
+	void add(const key_t& key, const value_t& value) {
+		double now = g_network->now();
```

Comment:
`now` is actually a global name defined in flow.h to return `g_network->now()`. In a case like this, we'd usually just use `now()` directly at the point of use, but if you want to create a local variable for it, it may be a good practice in general to avoid shadowing the global.